### ISSUE TITLE
Colors: compilation issues

### ISSUE THREAD

**Issue by hoshiryu** [STORM-IRIT/Radium-Engine#435]
I got 2 compilation issues with the Colors in my plugins:
 - When I use `Color::scatter` in my code, I get this error in `Color::scatter()`:
```
error: no match for ‘operator=’ (operand types are ‘__gnu_cxx::__alloc_traits<std::allocator<Ra::Core::Utils::ColorBase<float> >, Ra::Core::Utils::ColorBase<float> >::value_type’ {aka ‘Ra::Core::Utils::ColorBase<float>’} and ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<float, float>, const Eigen::CwiseBinaryOp<Eigen::internal::scalar_sum_op<float, float>, const Eigen::Matrix<float, 4, 1>, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, Eigen::Matrix<float, 4, 1> > >, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, const Eigen::Matrix<float, 4, 1> > >’)
                 color[i] = ( color[i] + ColorBase<_Scalar>::Constant( gamma ) ) * 0.5;
                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
 - When I use arithmetics on colors, I get this error:
```
error: no match for ‘operator=’ (operand types are ‘Ra::Core::Utils::Color’ {aka ‘Ra::Core::Utils::ColorBase<float>’} and ‘const Eigen::CwiseBinaryOp<Eigen::internal::scalar_sum_op<float, float>, const Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<float, float>, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, const Eigen::Matrix<float, 4, 1> >, const Eigen::Matrix<float, 4, 1> >, const Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<float, float>, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<float>, const Eigen::Matrix<float, 4, 1> >, const Eigen::Matrix<float, 4, 1> > >’)
     color = waves * color + ( 1 - waves ) * Ra::Core::Utils::Color::White();
                                                                           ^
```
This seems related to the `explicit` keyword in:
```c++
/// Copy constructor from Eigen expressions: `ColorBase<_Scalar> c( (*this) * 255 );`
template <typename Derived>
explicit inline ColorBase( const Eigen::MatrixBase<Derived>& v ) :
    VectorType( v.template cast<_Scalar>() ) {}
```
since when I remove it, all compiles nicely.

**Comment by MathiasPaulin** [comment]
This is an error, this `explicit` must not be there as it prevents automatic conversion from Eigen expression.
With this, the `test_core_color` does not compile.

**Comment by MathiasPaulin** [comment]
closed by PR #441
